File: Apps/BankBusiness/backend/model/PaymentAccount.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.model;

[Table("payment_account")]
public class PaymentAccount
{
    [Key, Column(name: "id"), DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id {get; set;}
    
    [Column(name: "name")]
    public string name {get; set;}
    
    [Column(name: "deposit")]
    public double deposit {get; set;}
    
    [ForeignKey("business_client_id")]
    public BusinessClient LinkedClient {get; set;}
    
    [Column("invoice_create_token")]
    public Guid? InvoiceCreateToken {get; set;}
}

File: Apps/BankBusiness/backend/src/main/scala/model/PaymentAccount.scala
package model

import io.getquill.{InsertMeta, SchemaMeta, UpdateMeta, insertMeta, schemaMeta, updateMeta}

import java.util.UUID

case class PaymentAccount(
                           id: Long,
                           name: String,
                           deposit: Double,
                           businessClientId: Long,
                           invoiceCreateToken: UUID
                         )

object PaymentAccount {
  inline given SchemaMeta[PaymentAccount] = schemaMeta(
    "payment_account",
    _.id -> "id",
    _.name -> "name",
    _.deposit -> "deposit",
    _.businessClientId -> "business_client_id",
    _.invoiceCreateToken -> "invoice_create_token"
  )

  inline given InsertMeta[PaymentAccount] = insertMeta(_.id)

  inline given UpdateMeta[PaymentAccount] = updateMeta(_.id)
}